content/kb/troubleshooting-volume-with-multipath.md

```diff
@@ -0,0 +1,108 @@
+E0416 08:49:27.567704 1 mount_linux.go:143] Mount failed: exit status 32
+Mounting command: mount
+Mounting arguments: -t ext4 -o defaults /dev/longhorn/pvc-d061512e-870a-4ece-bd45-2f04672d5256 /var/lib/kubelet/pods/cf0a0b5b-106e-4793-a74a-28bfae21be1a/volumes/kubernetes.io~csi/pvc-d061512e-870a-4ece-bd45-2f04672d5256/mount
+Output: mount: /var/lib/kubelet/pods/cf0a0b5b-106e-4793-a74a-28bfae21be1a/volumes/kubernetes.io~csi/pvc-d061512e-870a-4ece-bd45-2f04672d5256/mount: /dev/longhorn/pvc-d061512e-870a-4ece-bd45-2f04672d5256 already mounted or mount point busy.
+E0416 08:49:27.576477 1 mount_linux.go:487] format of disk "/dev/longhorn/pvc-d061512e-870a-4ece-bd45-2f04672d5256" failed: type:("ext4") target:("/var/lib/kubelet/pods/cf0a0b5b-106e-4793-a74a-28bfae21be1a/volumes/kubernetes.io~csi/pvc-d061512e-870a-4ece-bd45-2f04672d5256/mount") options:(["defaults"])error:(exit status 1)
+time="2020-04-16T08:49:27Z" level=error msg="GRPC error: rpc error: code = Internal desc = exit status 1"
+```
+
+## Details
+
+This is caused by that multipath is to create a multipath device for any eligible device path including every Longhorn volume device not explicitly blacklisted.
+
+### Troubleshooting
+
+1. Find the major:minor number of the Longhorn device. On the node, try `ls -l /dev/longhorn/`. The major:minor number will be shown as e.g. `8, 32` before the device name.
+
+```
+ls -l /dev/longhorn
+brw-rw---- 1 root root 8, 32 Aug 10 21:50 pvc-39c0db31-628d-41d0-b05a-4568ec02e487
+```
+
+2. Find what's the device generated by Linux for the same major:minor number. Use `ls -l /dev` and find the device for the same major:minor number, e.g. `/dev/sde`.
+
+```
+brw-rw---- 1 root disk      8,  32 Aug 10 21:50 sdc
+```
+
+3. Find the process. Use `lsof` to get the list of file handlers in use, then grep for the device name (e.g. `sde` or `/dev/longhorn/xxx`. You should find one there.
+
+```
+lsof | grep sdc
+multipath 514292                              root    8r      BLK               8,32        0t0        534 /dev/sdc
+multipath 514292 514297 multipath             root    8r      BLK               8,32        0t0        534 /dev/sdc
+multipath 514292 514299 multipath             root    8r      BLK               8,32        0t0        534 /dev/sdc
+multipath 514292 514300 multipath             root    8r      BLK               8,32        0t0        534 /dev/sdc
+multipath 514292 514301 multipath             root    8r      BLK               8,32        0t0        534 /dev/sdc
+multipath 514292 514302 multipath             root    8r      BLK               8,32        0t0        534 /dev/sdc
+multipath 514292 514304 multipath             root    8r      BLK               8,32        0t0        534 /dev/sdc
+```
+
+#### Solution
+
+To prevent multipath daemon from adding additional block devices created by Longhorn
```

Comment:
Follow these steps to prevent the multipath daemon from adding additional block devices created by Longhorn.

First check devices created by Longhorn using `lsblk`:

Reply:
changed